include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -293,11 +293,52 @@ struct __parallel_scan_submitter;
 template <typename _CustomName, typename... _PropagateScanName>
 struct __parallel_scan_submitter<_CustomName, __internal::__optional_kernel_name<_PropagateScanName...>>
 {
+    static constexpr int
+    __iters_per_item(::oneapi::dpl::__internal::__algorithm_type __algo_type)
+    {
+        using namespace ::oneapi::dpl::__internal;
+        switch (__algo_type)
+        {
+        case __algorithm_type::partition:
+            return 1;
+        case __algorithm_type::unique:
+            return 2;
+        case __algorithm_type::set_operation:
+            return 4;
+        case __algorithm_type::scan_copy:
+            return 16;
+        case __algorithm_type::transform_scan:
+            return 32;
+        default:
+            return 16;
+        }
+    }
+
+    static constexpr ::std::size_t
+    __preferred_wgroup_size(::oneapi::dpl::__internal::__algorithm_type __algo_type)
+    {
+        using namespace ::oneapi::dpl::__internal;
+        switch (__algo_type)
+        {
+        case __algorithm_type::transform_scan:
+            return 128;
+        case __algorithm_type::scan_copy:
+            return 256;
+        case __algorithm_type::unique:
+        case __algorithm_type::set_operation:
+            return 512;
+        case __algorithm_type::partition:
+        default:
+            return ::std::numeric_limits<::std::size_t>::max();
```

Comment:
Could you please explain the goal of return ```::std::numeric_limits<::std::size_t>::max()``` ?

Reply:
The idea here is that the preferred work-group size is the largest allowed by the device. It is clipped down using `std::min` after this call.